### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by kov**
I started implementing the improvements to readability that we discussed, but ended up deciding that it was better to first add support for several versions of the benchmark first, so we start from a more solid foundation. This change needs to be applied in lock-step with: https://github.com/iocost-benchmark/iocost-benchmarks/pull/24

### CODE REVIEW

File: .github/workflows/build-resctl-demo.yml [hunk 1]
```diff
@@ -6,13 +6,14 @@ on:
...
   # every Monday at 01:00
+  push:
   schedule:
   - cron: "0 1 * * 1"
   workflow_dispatch:
 
 jobs:
-  run:
-    name: Build resctl-demo
+  build_2_1:
```
Comment (obbardc) [inline .github/workflows/build-resctl-demo.yml]: could this not be done with a matrix instead ?
https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

**Review by obbardc — CHANGES_REQUESTED** [review]
small nit about using a matrix

File: .github/workflows/build-resctl-demo.yml (hunk 1 above)
Comment (kov) [reply .github/workflows/build-resctl-demo.yml]: Great idea!

File: .github/workflows/build-resctl-demo.yml [hunk 2]
```diff
@@ -6,14 +6,19 @@ on:
...
 
 jobs:
-  run:
+  build:
     name: Build resctl-demo
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version:
+          [{ minor: "v2.1", ref: "v2.1.2" }, { minor: "v2.2", ref: "v2.2.0" }]
```
Comment (obbardc) [inline .github/workflows/build-resctl-demo.yml]:
```suggestion
          - { minor: "v2.1", ref: "v2.1.2" }
          - { minor: "v2.2", ref: "v2.2.0" }
```
would syntax like this make it more readable ?

**Review by obbardc — CHANGES_REQUESTED** [review]

File: .github/workflows/build-resctl-demo.yml (hunk 2 above)
Comment (kov) [reply .github/workflows/build-resctl-demo.yml]: Much more readable indeed!

**Review by obbardc — APPROVED** [review]

**Pull request merged**